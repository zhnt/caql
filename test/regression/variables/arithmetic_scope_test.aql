// Arithmetic expressions with complex scoping
let a = 2
let b = 3
{
    let c = a * b  // 2 * 3 = 6
    a = a + c      // 2 + 6 = 8
    {
        let a = 10  // shadow outer a
        let d = a + b + c  // 10 + 3 + 6 = 19
        b = b * 2   // modify outer b: 3 * 2 = 6
        print("inner: a=" + a + ", b=" + b + ", c=" + c + ", d=" + d)
    }
    print("middle: a=" + a + ", b=" + b + ", c=" + c)
}
print("outer: a=" + a + ", b=" + b)

