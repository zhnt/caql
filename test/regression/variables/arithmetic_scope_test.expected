inner: a=10, b=6, c=6, d=19
middle: a=8, b=6, c=6
outer: a=8, b=6

